assets/src/stories-editor/components/custom-video-block-edit.js

```diff
@@ -341,25 +357,62 @@ CustomVideoBlockEdit.propTypes = {
 	videoFeaturedImage: PropTypes.shape( {
 		source_url: PropTypes.string,
 	} ),
+	uploadVideoFrame: PropTypes.func,
 };
 
 export default compose( [
-	withSelect( ( select, props ) => {
+	withDispatch( ( dispatch ) => {
+		const { saveMedia } = dispatch( 'core' );
+
+		return {
+			saveMedia,
+		};
+	} ),
+	withSelect( ( select, { attributes, setAttributes, saveMedia } ) => {
 		const { getMedia } = select( 'core' );
 		const { getSettings } = select( 'core/block-editor' );
-		const { __experimentalMediaUpload } = getSettings();
+		const { __experimentalMediaUpload: mediaUpload } = getSettings();
 
 		let videoFeaturedImage;
 
-		if ( props.attributes.id && ! props.attributes.poster ) {
-			const media = getMedia( props.attributes.id );
-			const featuredImage = media && get( media, [ '_links', 'wp:featuredmedia', 0, 'href' ], null );
-			videoFeaturedImage = featuredImage && getMedia( Number( featuredImage.split( '/' ).pop() ) );
+		const { id, poster } = attributes;
+
+		if ( id && ! poster ) {
+			const media = getMedia( id );
+			videoFeaturedImage = media && media.featured_media > 0 && getMedia( media.featured_media );
```

Comment:
It looks like by the time when `videoFeaturedImage` is received, getting the first frame of the video has already run, meaning that a new image is created even though a previous image exists already. Is it using the existing image properly when testing?

Reply:
You're right, it does upload it again. 😞 I'll look into it.